pyccel/parser/syntactic.py

```diff
@@ -494,10 +498,23 @@ def _visit_BooleanOperatorNode(self, stmt):
 
         first = self._visit(stmt.first)
         second = self._visit(stmt.second)
+
+        assoc  = isinstance(stmt.second, AssociativeParenthesisNode)
+
         if stmt.value == 'and':
-            return And(first, second, evaluate=False)
+            if not assoc and isinstance(second, PyccelOr):
+                args  = second.args
+                first = PyccelAnd(first, args[0]  )
+                return  PyccelOr (first, *args[1:])
```

Comment:
How do you know that a `PyccelOr` is required?

Reply:
because we have two cases in `BooleanOperatorNode` either `and` or `or`

Reply:
Yes, but the user might put brackets in even if it's not necessary

Reply:
okay i will make it general

Reply:
the brackets case is handled in the else condition, this condition enables us to evaluate the expression from left to right instead of doing it from right to left